Update xapian version

I'm sorry but it's becoming a pain to test. I can't compile on macOS and neither can I on a `debian:bullseye` container

```
[4/102] ccache c++ -Itest/library.p -Iinclude -I../../SOURCE/libkiwix/include -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -DCURL_STATICLIB -pthread -isystem../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest -isystemsubprojects/googletest-release-1.8.1/googletest -isystem../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest/include -MD -MQ test/library.p/.._subprojects_googletest-release-1.8.1_googletest_src_gtest_main.cc.o -MF test/library.p/.._subprojects_googletest-release-1.8.1_googletest_src_gtest_main.cc.o.d -o test/library.p/.._subprojects_googletest-release-1.8.1_googletest_src_gtest_main.cc.o -c ../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest/src/gtest_main.cc
[5/102] ccache c++ -Itest/regex.p -Iinclude -I../../SOURCE/libkiwix/include -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -DCURL_STATICLIB -pthread -isystem../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest -isystemsubprojects/googletest-release-1.8.1/googletest -isystem../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest/include -MD -MQ test/regex.p/regex.cpp.o -MF test/regex.p/regex.cpp.o.d -o test/regex.p/regex.cpp.o -c ../../SOURCE/libkiwix/test/regex.cpp
[6/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/meson-generated_.._.._static_libkiwix-resources.cpp.o -MF src/libkiwix.a.p/meson-generated_.._.._static_libkiwix-resources.cpp.o.d -o src/libkiwix.a.p/meson-generated_.._.._static_libkiwix-resources.cpp.o -c static/libkiwix-resources.cpp
[7/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/meson-generated_.._.._static_libkiwix-i18n-resources.cpp.o -MF src/libkiwix.a.p/meson-generated_.._.._static_libkiwix-i18n-resources.cpp.o.d -o src/libkiwix.a.p/meson-generated_.._.._static_libkiwix-i18n-resources.cpp.o -c static/libkiwix-i18n-resources.cpp
[8/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/bookmark.cpp.o -MF src/libkiwix.a.p/bookmark.cpp.o.d -o src/libkiwix.a.p/bookmark.cpp.o -c ../../SOURCE/libkiwix/src/bookmark.cpp
[9/102] ccache c++ -Itest/library.p -Iinclude -I../../SOURCE/libkiwix/include -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -DCURL_STATICLIB -pthread -isystem../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest -isystemsubprojects/googletest-release-1.8.1/googletest -isystem../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest/include -MD -MQ test/library.p/library.cpp.o -MF test/library.p/library.cpp.o.d -o test/library.p/library.cpp.o -c ../../SOURCE/libkiwix/test/library.cpp
[10/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/book.cpp.o -MF src/libkiwix.a.p/book.cpp.o.d -o src/libkiwix.a.p/book.cpp.o -c ../../SOURCE/libkiwix/src/book.cpp
[11/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/manager.cpp.o -MF src/libkiwix.a.p/manager.cpp.o.d -o src/libkiwix.a.p/manager.cpp.o -c ../../SOURCE/libkiwix/src/manager.cpp
[12/102] ccache c++ -Itest/library.p -Iinclude -I../../SOURCE/libkiwix/include -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -DCURL_STATICLIB -pthread -isystem../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest -isystemsubprojects/googletest-release-1.8.1/googletest -isystem../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest/include -MD -MQ test/library.p/.._subprojects_googletest-release-1.8.1_googletest_src_gtest-all.cc.o -MF test/library.p/.._subprojects_googletest-release-1.8.1_googletest_src_gtest-all.cc.o.d -o test/library.p/.._subprojects_googletest-release-1.8.1_googletest_src_gtest-all.cc.o -c ../../SOURCE/libkiwix/subprojects/googletest-release-1.8.1/googletest/src/gtest-all.cc
[13/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/libxml_dumper.cpp.o -MF src/libkiwix.a.p/libxml_dumper.cpp.o.d -o src/libkiwix.a.p/libxml_dumper.cpp.o -c ../../SOURCE/libkiwix/src/libxml_dumper.cpp
[14/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/opds_dumper.cpp.o -MF src/libkiwix.a.p/opds_dumper.cpp.o.d -o src/libkiwix.a.p/opds_dumper.cpp.o -c ../../SOURCE/libkiwix/src/opds_dumper.cpp
[15/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/downloader.cpp.o -MF src/libkiwix.a.p/downloader.cpp.o.d -o src/libkiwix.a.p/downloader.cpp.o -c ../../SOURCE/libkiwix/src/downloader.cpp
[16/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/library_dumper.cpp.o -MF src/libkiwix.a.p/library_dumper.cpp.o.d -o src/libkiwix.a.p/library_dumper.cpp.o -c ../../SOURCE/libkiwix/src/library_dumper.cpp
[17/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/html_dumper.cpp.o -MF src/libkiwix.a.p/html_dumper.cpp.o.d -o src/libkiwix.a.p/html_dumper.cpp.o -c ../../SOURCE/libkiwix/src/html_dumper.cpp
[18/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/subprocess.cpp.o -MF src/libkiwix.a.p/subprocess.cpp.o.d -o src/libkiwix.a.p/subprocess.cpp.o -c ../../SOURCE/libkiwix/src/subprocess.cpp
[19/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/server.cpp.o -MF src/libkiwix.a.p/server.cpp.o.d -o src/libkiwix.a.p/server.cpp.o -c ../../SOURCE/libkiwix/src/server.cpp
[20/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/tools_base64.cpp.o -MF src/libkiwix.a.p/tools_base64.cpp.o.d -o src/libkiwix.a.p/tools_base64.cpp.o -c ../../SOURCE/libkiwix/src/tools/base64.cpp
[21/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/aria2.cpp.o -MF src/libkiwix.a.p/aria2.cpp.o.d -o src/libkiwix.a.p/aria2.cpp.o -c ../../SOURCE/libkiwix/src/aria2.cpp
[22/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/library.cpp.o -MF src/libkiwix.a.p/library.cpp.o.d -o src/libkiwix.a.p/library.cpp.o -c ../../SOURCE/libkiwix/src/library.cpp
FAILED: src/libkiwix.a.p/library.cpp.o
ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/library.cpp.o -MF src/libkiwix.a.p/library.cpp.o.d -o src/libkiwix.a.p/library.cpp.o -c ../../SOURCE/libkiwix/src/library.cpp
In file included from /kiwix-build/./BUILD_native_static/INSTALL/include/xapian.h:80,
                 from ../../SOURCE/libkiwix/src/library.cpp:37:
/kiwix-build/./BUILD_native_static/INSTALL/include/xapian/queryparser.h: In member function ‘void Xapian::QueryParser::add_valuerangeprocessor(Xapian::ValueRangeProcessor*)’:
/kiwix-build/./BUILD_native_static/INSTALL/include/xapian/queryparser.h:1266:26: error: ‘void Xapian::QueryParser::add_valuerangeprocessor(Xapian::ValueRangeProcessor*)’ is deprecated [-Werror=deprecated-declarations]
 1266 |  add_rangeprocessor((new ShimRangeProcessor(vrproc))->release());
      |                          ^~~~~~~~~~~~~~~~~~
In file included from /kiwix-build/./BUILD_native_static/INSTALL/include/xapian/types.h:28,
                 from /kiwix-build/./BUILD_native_static/INSTALL/include/xapian.h:47,
                 from ../../SOURCE/libkiwix/src/library.cpp:37:
/kiwix-build/./BUILD_native_static/INSTALL/include/xapian/queryparser.h:1246:28: note: declared here
 1246 |     XAPIAN_DEPRECATED(void add_valuerangeprocessor(Xapian::ValueRangeProcessor * vrproc)) {
      |                            ^~~~~~~~~~~~~~~~~~~~~~~
/kiwix-build/./BUILD_native_static/INSTALL/include/xapian/deprecated.h:50:32: note: in definition of macro ‘XAPIAN_DEPRECATED’
   50 | #  define XAPIAN_DEPRECATED(D) D __attribute__((__deprecated__))
      |                                ^
cc1plus: all warnings being treated as errors
[23/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/tools_regexTools.cpp.o -MF src/libkiwix.a.p/tools_regexTools.cpp.o.d -o src/libkiwix.a.p/tools_regexTools.cpp.o -c ../../SOURCE/libkiwix/src/tools/regexTools.cpp
[24/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/tools_pathTools.cpp.o -MF src/libkiwix.a.p/tools_pathTools.cpp.o.d -o src/libkiwix.a.p/tools_pathTools.cpp.o -c ../../SOURCE/libkiwix/src/tools/pathTools.cpp
[25/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/tools_stringTools.cpp.o -MF src/libkiwix.a.p/tools_stringTools.cpp.o.d -o src/libkiwix.a.p/tools_stringTools.cpp.o -c ../../SOURCE/libkiwix/src/tools/stringTools.cpp
[26/102] ccache c++ -Isrc/libkiwix.a.p -Isrc -I../../SOURCE/libkiwix/src -Iinclude -I../../SOURCE/libkiwix/include -Istatic -I/kiwix-build/./BUILD_native_static/INSTALL/include -I/kiwix-build/BUILD_native_static/INSTALL/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c++11 -O0 -g -fPIC -DCURL_STATICLIB -pthread -MD -MQ src/libkiwix.a.p/search_renderer.cpp.o -MF src/libkiwix.a.p/search_renderer.cpp.o.d -o src/libkiwix.a.p/search_renderer.cpp.o -c ../../SOURCE/libkiwix/src/search_renderer.cpp
ninja: build stopped: subcommand failed.
```

_Originally posted by @rgaudin in https://github.com/kiwix/libkiwix/issues/940#issuecomment-1518730634_

--- a/kiwixbuild/dependencies/xapian.py
+++ b/kiwixbuild/dependencies/xapian.py
@@ -12,8 +12,9 @@ class Xapian(Dependency):
     name = "xapian-core"
 
     class Source(ReleaseDownload):
-        archive = Remotefile('xapian-core-1.4.18.tar.xz',
-                             '196ddbb4ad10450100f0991a599e4ed944cbad92e4a6fe813be6dce160244b77')
+        archive = Remotefile('xapian-core-1.4.22.tar.xz',
+                             '05884af00b06702ce486057d62a3bfbe6606cf965ada0f5ea570b328a2fa1ea8')
+        patches = ['xapian_win32.patch']
 
     class Builder(MakeBuilder):
         configure_option = "--disable-sse --disable-backend-chert --disable-backend-remote --disable-documentation"

--- a/kiwixbuild/patches/xapian_win32.patch
+++ b/kiwixbuild/patches/xapian_win32.patch
@@ -1,0 +1,20 @@
+diff '--color=auto' -ur xapian-core-1.4.22/common/safesyssocket.h xapian-core-1.4.22.patched/common/safesyssocket.h
+--- xapian-core-1.4.22/common/safesyssocket.h	2023-02-02 02:00:40.000000000 +0100
++++ xapian-core-1.4.22.patched/common/safesyssocket.h	2023-04-25 15:46:52.228256033 +0200
+@@ -27,6 +27,7 @@
+ # include <sys/socket.h>
+ #else
+ # include "safewinsock2.h"
++# include <ws2tcpip.h>
+ #endif
+ 
+ #ifdef __WIN32__
+@@ -64,7 +65,7 @@
+ # endif
+ # define socket(D,T,P) socket_(D,T,P)
+ 
+-inline int accept_(int sockfd, struct sockaddr* addr, SOCKLEN_T* addrlen) {
++inline int accept_(int sockfd, struct sockaddr* addr, socklen_t* addrlen) {
+     // Winsock2's accept() returns the unsigned type SOCKET, which is a 32-bit
+     // type for WIN32 and a 64-bit type for WIN64.
+     //

--- a/kiwixbuild/versions.py
+++ b/kiwixbuild/versions.py
@@ -39,7 +39,7 @@ release_versions = {
 
 # This is the "version" of the whole base_deps_versions dict.
 # Change this when you change base_deps_versions.
-base_deps_meta_version = '84'
+base_deps_meta_version = '85'
 
 base_deps_versions = {
   'zlib' : '1.2.12',
@@ -47,7 +47,7 @@ base_deps_versions = {
   'zstd' : '1.5.2',
   'docoptcpp' : '0.6.2',
   'uuid' : '1.43.4',
-  'xapian-core' : '1.4.18',
+  'xapian-core' : '1.4.22',
   'mustache' : '4.1',
   'pugixml' : '1.2',
   'libmicrohttpd' : '0.9.76',

